lib/dice_bag/available_templates.rb

```diff
@@ -40,35 +38,33 @@ def inherited(base)
         template_checkers << base
       end
 
-      def all(gem_names=[])
-        #all the classes than inherit from us in the ruby runtime
+      def all(gem_names = [])
+        # all the classes than inherit from us in the ruby runtime
         available_templates = []
 
         template_checkers.each do |template_checker|
           checker = template_checker.new
           next if !gem_names.empty? && !checker_within_given_gems?(checker, gem_names)
+
           location = checker.templates_location
           checker.templates.each do |template, save_as|
-            available_templates.push( DefaultTemplateFile.new(template, location, save_as) )
+            available_templates.push(DefaultTemplateFile.new(template, location, save_as))
           end
         end
         available_templates
       end
 
       def template_filename_for(filename)
-        self.all.each do |template|
+        all.each do |template|
           if template.filename.include? filename
-            return template.file
+            template.file
```

Comment:
wouldn't this change the behavior slightly? with return it would complete fast and return the expected result, without it would go through the whole list and return the last iteration's result.
Can this can be refactored to use `find`?

Reply:
Good catch! Indeed looks like a perfect fit for `find` 9ade7ec